include/git2/common.h

```diff
@@ -59,6 +59,12 @@ typedef size_t size_t;
 #define GIT_WIN32 1
 #endif
 
+#if defined(__has_feature)
+#if __has_feature(address_sanitizer)
+#define GIT_ASAN 1
```

Comment:
The feature is enabled when AddressSanitizer is available and currently being used for the current compile run.

The CMake modules take care of automatically passing in the correct flags to the compiler when you `-DASAN=ON`, which will in turn cause that define to exist. The ultimate goal is to be able to tell at compile-time whether ASan is enabled or not (because the modules don't set anything since there's a clang feature macro for it), so we can skip the oom test below (e.g. doing weird things that makes ASan unhappy, like allocating the whole memory). That's the only point of it actually, so if having that in `common.h` feels wrong, I can move it somewhere else.

I don't really get the point about providing functionality. Because ASan incurs a [typical 2x slowdown](http://clang.llvm.org/docs/AddressSanitizer.html), it's not something I expect to be queried at runtime. It can be used as a debugging tool (build libgit2 with ASan, link your app with that version, run and wait).